### PR TITLE
ath79: add support for Belkin F9K1115V2

### PR DESCRIPTION
**Pull request by xdarklight**
This adds support for the Belkin F9K1115V2
It was a supported device in the ar71xx target but has not been ported to ath79 so far.

Support is based on F9K1115 v2 from PR #3380
Many thanks to all who helped to get this in good shape and especially @dmascord for working on that!

There's different ways to write F9K1115V2 (with blank, lower-/upper-case "v" in v2, ...)
I decided to go with what's on the label of the device: F9K1115V2
This can be seen in one of the FCC photos too: https://fccid.io/K7SF9K1115V2/Label/Label-Location-2031947

### CODE REVIEW

**Comment by adschm** [comment]
> qca9558_belkin_f9xyyy-v2.dtsi

This one is missing one "y". Apart from that, I don't really think it's too nice.

Maybe just belkin_f9.dtsi or belkin_f9x-v2.dtsi? Or is there a common board or something that may be more appropriate?

File: target/linux/ath79/image/generic.mk [hunk 1]
```diff
@@ -402,6 +402,27 @@ define Device/belkin_f9j1108-v2
 endef
 TARGET_DEVICES += belkin_f9j1108-v2
 
+define Device/belkin_f9k1115v2
```
Comment (adschm) [inline target/linux/ath79/image/generic.mk:405]: Please separate v2 with a hyphen here, like for the other device.

File: target/linux/ath79/image/generic.mk [hunk 2]
```diff
@@ -402,6 +402,27 @@ define Device/belkin_f9j1108-v2
...
+  LOADER_FLASH_OFFS := 0x50000
+  COMPILE := loader-$(1).bin loader-$(1).uImage
+  COMPILE/loader-$(1).bin := loader-okli-compile
+  COMPILE/loader-$(1).uImage := append-loader-okli $(1) | pad-to 64k | \
+	lzma | uImage lzma
+  KERNEL := kernel-bin | append-dtb | lzma | uImage lzma -M 0x4f4b4c49
+  IMAGES += factory.bin
+  IMAGE/factory.bin := append-kernel | pad-to $$$$(BLOCKSIZE) | \
+	append-rootfs | pad-rootfs | check-size | \
+	edimax-headers eDiMaX F9K1115V2 | pad-to $$$$(BLOCKSIZE)
```
Comment (adschm) [inline target/linux/ath79/image/generic.mk:422]: That's a lot of duplicate definition, please create a shared one (there should be enough examples in the file).

**Comment by adschm** [comment]
TBH, I don't think we really need the two commits here. But I only have a 70/30 tendency ...

File: target/linux/ath79/image/generic.mk (hunk 2 above)
Comment (xdarklight) [reply target/linux/ath79/image/generic.mk:422]: OK, I will consolidate this.
do you prefer `Device/belkin_f9x_v2_common` or `Device/belkin_edimax_headers` (or something else)?

File: target/linux/ath79/image/generic.mk (hunk 1 above)
Comment (xdarklight) [reply target/linux/ath79/image/generic.mk:405]: just to double-check: you still prefer to have the dash (to make it consistent with other boards) even the device label (as state in the PR description) does not use a dash/space?

**Comment by xdarklight** [comment]
> > qca9558_belkin_f9xyyy-v2.dtsi
> 
> This one is missing one "y". Apart from that, I don't really think it's too nice.

oh, good point

> Maybe just belkin_f9.dtsi or belkin_f9x-v2.dtsi? Or is there a common board or something that may be more appropriate?

I'll go with `belkin_f9x-v2.dtsi` as it shows better that there's some placeholder.
And also it has v2 in it to clarify that it's not any f9... board

File: target/linux/ath79/image/generic.mk (hunk 2 above)
Comment (adschm) [reply target/linux/ath79/image/generic.mk:422]: The former. You might even drop the common, just using Device/belkin_f9x_v2 or similar.

Experience tells that being too general doesn't help here (unless you have other devices to join as well, I haven't checked).

File: target/linux/ath79/image/generic.mk (hunk 1 above)
Comment (adschm) [reply target/linux/ath79/image/generic.mk:405]: I'm under the impression that the v2 represents a version/variant. In that case, it makes sense to separate it as we then could have a separate version field for metadata later without having to touch the name again. If that's not the case, and the v2 stands for something entirely different, one might reconsider.

Comment (xdarklight) [reply target/linux/ath79/image/generic.mk:405]: according to their download page this really is "version 2.0" hardware: https://www.belkin.com/us/support-article?articleNum=42287
it seems like v1 never made to the public.

I'll update the naming to be -v2 as suggested

**Comment by xdarklight** [comment]
v2 of this pull request. changes since v1:
* changed naming to F9K1115 v2 (image building recipe, .dtsi filename, etc.)
* use common code for the image building recipe between F9K1115 v2 and F9J1108 v2

**Comment by adschm** [comment]
Thanks, will be merged later today with minor adjustments. Do we really need that `(AC1750 DB Wi-Fi)` in the name, particularly since it's the same for both devices, or could we outsource that to the Wiki page and remove in the code?

**Comment by xdarklight** [comment]
> Do we really need that (AC1750 DB Wi-Fi) in the name

I only added it for consistency with other boards. it's the name which is printed on the box
no strong feelings either way from me, so if you decide to drop or keep it: either way is fine with me.

**Pull request closed without merging**